### PR TITLE
Upgrade to manifest v3

### PR DESCRIPTION
**Pull request by madhur**
@benjaminoakes  Pls check the initial PR for v3 manifest upgrade. I have verified the happy paths to be working in various browsers such as Chrome, Brave.

Few points:
* The alert popup is no longer supported in background scripts. I have instead relied on installed event to show the tutorial in new tab.

* Added the gulp for automating the build tasks and webpack to bundle the javascripts.

Steps to run:
1. Do `npm install` on the upgrade branch and install `npm install -g gulp-cli`
2. run `gulp`
3. This should output `dist` folder which can be directly loaded as unpacked extension into browser.

Development mode:
1. Instead of `gulp` , run `gulp watch`, this runs gulp in watch mode where any changes in `src` folders will be picked up and reflected in `dist` folder immediately.

Other minor changes:
* Renamed `ext` to `dist`
* Changed `chrome.tabs.getAllInWindow` in favor of `chrome.tabs.query`

Further improvments (which can be added but didn't seem urgent as of now):
* I saw few jasmine tests, but was not able to run it in my version due to unmocked chrome API's.
* We can add html and js minifications in the production gener.

### CODE REVIEW

**Comment by madhur** [comment]
Related to #49

File: src/javascripts/background.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
+
+/**
+ * Chrome plugin to cycle through tabs.
+ * 
+ * @author Benjamin Oakes <hello@benjaminoakes.com>, @benjaminoakes
```
Comment (benjaminoakes) [inline src/javascripts/background.js:5]: Seems like you wrote a lot of this @madhur :smile:

Let's make sure to give you proper attribution as well

**Review by benjaminoakes — APPROVED** [review]
Wow, this is great work!  Thank you!

I appreciate your consideration of [my thoughts on JSDoc vs TypeScript](https://github.com/TabCarousel/TabCarousel/issues/49#issuecomment-1847319687).  That said, if you do want to introduce TypeScript in the future, please feel free to do so.

A couple of things stand out:

This is a pull request against `master`.  For better or worse, this repository uses the [Git Flow](https://nvie.com/posts/a-successful-git-branching-model/) branching model, so I think it should be merged into `develop` and then merged to master when released to the Chrome Web Store.  I honestly haven't done this in quite a while, but that's my recollection of the process.  :smile: 

Also:

> **Continuous integration has not been set up**
> GitHub Actions and several other apps can be used to automatically catch bugs and enforce style. 

You're correct that the deleted Jasmine specs didn't add much value.  What do you think about adding something in CI, perhaps in a future PR?  It would be nice to try building the extension, checking style, etc.  (Issue: https://github.com/TabCarousel/TabCarousel/issues/52)

File: src/javascripts/background.js (hunk 1 above)
Comment (benjaminoakes) [reply src/javascripts/background.js:5]: Added

**Comment by benjaminoakes** [comment]
> This is a pull request against `master`.  For better or worse, this repository uses the [Git Flow](https://nvie.com/posts/a-successful-git-branching-model/) branching model, so I think it should be merged into `develop` and then merged to master when released to the Chrome Web Store.  I honestly haven't done this in quite a while, but that's my recollection of the process.  :smile: 

I attempted to change the base branch to `develop` so that this can merge and ran into conflicts

> **This branch has conflicts that must be resolved**
> 
> **Conflicting files**
> 
> ```
> .gitignore
> README.md
> ext/background.html
> ext/options.html
> package.json 
> ```

How would you like to proceed?  Attempt to resolve the conflicts or reconsider the branching strategy?  Honestly, I've barely touched this code in ages, so I can go either way, but if we're not going to use `develop` anymore, we shouldn't have it be the default branch any longer.

Please let me know your preference and how I can be of help.

**Comment by madhur** [comment]
> > This is a pull request against `master`.  For better or worse, this repository uses the [Git Flow](https://nvie.com/posts/a-successful-git-branching-model/) branching model, so I think it should be merged into `develop` and then merged to master when released to the Chrome Web Store.  I honestly haven't done this in quite a while, but that's my recollection of the process.  😄
> 
> I attempted to change the base branch to `develop` so that this can merge and ran into conflicts
> 
> > **This branch has conflicts that must be resolved**
> > **Conflicting files**
> > ```
> > .gitignore
> > README.md
> > ext/background.html
> > ext/options.html
> > package.json 
> > ```
> 
> How would you like to proceed? Attempt to resolve the conflicts or reconsider the branching strategy? Honestly, I've barely touched this code in ages, so I can go either way, but if we're not going to use `develop` anymore, we shouldn't have it be the default branch any longer.
> 
> Please let me know your preference and how I can be of help.

Have resolved the conflicts and merged the PR. The branching strategy looks fine.

**Pull request merged**